lib/merkle_tree/src/with_version.rs

```diff
@@ -12,6 +12,11 @@ pub struct MerkleTreeVersion<DB: Database = RocksDBWrapper, P: TreeParams = Defa
     pub block: u64,
 }
 
+pub struct BlockMerkleTreeData {
+    pub block_start: MerkleTreeVersion,
+    pub block_end: MerkleTreeVersion,
+}
```

Comment:
This does not belong to `merkle_tree` either. It's way more high-level than what `merkle_tree` operates on.